### PR TITLE
feat: report machine bootstrap failures

### PR DESCRIPTION
**Pull request by asherdale**
- capture startup-script and daemon-install failures during managed machine bootstrap through a daemon-authenticated endpoint
- persist the latest failure on the machine as structured JSON with stage, exit and capture status, a bounded output tail, truncation state, and server timestamp
- expose bootstrap failures through `inspect_machine`, clear stale failures when a pool machine is claimed again, and add `inspect_machine` and `list_machines` to the frontend machine toolset
- warn users about Unikraft startup-script size constraints while keeping the bootstrap launcher compact across Unikraft, Blaxel, and Daytona

### CODE REVIEW

File: internal/httpapi/machine_daemon_routes.go [hunk 1]
```diff
@@ -54,6 +54,41 @@ func (s strictOpenAPIServer) BootstrapDaemon(
...
+		ctx,
+		executionstore.MachineBootstrapFailureInput{
+			OrgID:           scope.OrgID,
+			MachineID:       scope.MachineID,
+			DaemonTokenID:   scope.DaemonTokenID,
+			Stage:           string(request.Params.Stage),
+			ExitStatus:      request.Params.ExitStatus,
+			OutputTail:      outputTail,
+			OutputTruncated: outputTruncated,
+		},
```
[the remark this answers is not in the document]
Comment (asherdale) [reply internal/httpapi/machine_daemon_routes.go]: nah this is intentional - `capture_status` is just an internal status from the output capture pipeline, not another bootstrap exit status. we collapse it into `output_truncated` instead of storing another implementation detail on every failure

File: internal/httpapi/machine_daemon_routes.go [hunk 2]
```diff
@@ -54,6 +54,41 @@ func (s strictOpenAPIServer) BootstrapDaemon(
...
+	scope, scopeErr := machineDaemonScopeFromContext(ctx)
+	if scopeErr != nil {
+		return nil, *scopeErr
+	}
+	var outputTail []byte
+	if request.Body != nil {
+		outputTail = []byte(*request.Body)
+	}
+	outputTruncated := request.Params.CaptureStatus != 0 ||
+		len(outputTail) > executionstore.MaxMachineBootstrapFailureOutputBytes
```
[the remark this answers is not in the document]
Comment (asherdale) [reply internal/httpapi/machine_daemon_routes.go]: this is intentional too - if `capture_status` is nonzero, we cant know whether we actually captured the full output, even if the body is short or empty. marking it as `output_truncated` is the safer behavior and we have a test for this exact background process edge case

File: internal/httpapi/machine_daemon_routes.go [hunk 3]
```diff
@@ -54,6 +54,41 @@ func (s strictOpenAPIServer) BootstrapDaemon(
 	return openapi.BootstrapDaemon200JSONResponse(response), nil
 }
 
+func (s strictOpenAPIServer) RecordDaemonBootstrapFailure(
+	ctx context.Context,
+	request openapi.RecordDaemonBootstrapFailureRequestObject,
+) (openapi.RecordDaemonBootstrapFailureResponseObject, error) {
```
Comment (cmsparks) [inline internal/httpapi/machine_daemon_routes.go]: can we add a new wide log struct in logent to attach some of the failure output to our wide events in this route?

would make it easier to search across common errors

**Comment by asherdale** [comment]
CI passed here: https://github.com/omnara-ai/omnara/actions/runs/31130945589

File: internal/machinepool/providers/managed_daemon_launcher.go [hunk 4]
```diff
@@ -59,27 +59,28 @@ func ValidateManagedStartupScript(what, startupScript string) error {
...
+func ManagedBootScriptPayload() string {
+	return base64.StdEncoding.EncodeToString([]byte(ManagedBootScript()))
 }
 
 func ManagedDaemonLauncherArgs() []string {
 	return []string{
 		"/bin/sh",
 		"-c",
-		`original_umask=$(umask);umask${IFS}077;bootstrap=/tmp/omnarad-bootstrap;printf${IFS}%s${IFS}${OMNARA_BOOTSTRAP_SCRIPT:?}|base64${IFS}-d>$bootstrap&&unset${IFS}OMNARA_BOOTSTRAP_SCRIPT&&umask${IFS}"$original_umask"&&exec${IFS}/bin/sh${IFS}$bootstrap`,
+		`m=$(umask);umask${IFS}077;b=/tmp/omnarad-bootstrap;printf${IFS}%s${IFS}${OMNARA_BOOTSTRAP_SCRIPT:?}|base64${IFS}-d>$b&&unset${IFS}OMNARA_BOOTSTRAP_SCRIPT&&umask${IFS}"$m"&&exec${IFS}/bin/sh${IFS}$b`,
```
Comment (cmsparks) [inline internal/machinepool/providers/managed_daemon_launcher.go]: can we clean this up, it's pretty hard to read

File: migrations/000010_machine_bootstrap_failure.sql [hunk 5]
```diff
@@ -0,0 +1,10 @@
+-- +goose Up
+ALTER TABLE machines
+ADD COLUMN bootstrap_failure jsonb,
```
Comment (cmsparks) [inline migrations/000010_machine_bootstrap_failure.sql]: Should this be a more generic column without the constraint? I know currently bootstrap failure is somewhat specific to the machine pool, but is it possible that we'd want to store certain byo machine daemon failures here, or no?

**Review by cmsparks — APPROVED** [review]

File: internal/httpapi/machine_daemon_routes.go (hunk 3 above)
Comment (asherdale) [reply internal/httpapi/machine_daemon_routes.go]: good call, added `logent.MachineFailureReport` here

File: internal/machinepool/providers/managed_daemon_launcher.go (hunk 4 above)
Comment (asherdale) [reply internal/machinepool/providers/managed_daemon_launcher.go]: splitted this up so it's more readable, but the minimal script itself is necessary because unikraft has strict limits on the initial payload

File: migrations/000010_machine_bootstrap_failure.sql (hunk 5 above)
Comment (asherdale) [reply migrations/000010_machine_bootstrap_failure.sql]: agreed, renamed the column to `failure_report` and removed the `source_kind = 'pool'` restriction
